### DISCUSSION TITLE
Bug: Indent rule inconsistent with Prettier ESLint

### DISCUSSION THREAD

**Discussion by justingolden21** [eslint/eslint#15866] [Q&A]
### Environment

Node version: v16.13.0
npm version: v8.1.0
Local ESLint version: v7.32.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19044

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
	"env": {
		"browser": true,
		"es2021": true,
		"node": true
	},
	"extends": ["airbnb-base", "prettier"],
	"parserOptions": {
		"ecmaVersion": 2019,
		"sourceType": "module"
	},
	"plugins": ["prettier", "svelte3"],
	"overrides": [
		{
			"files": ["*.svelte"],
			"processor": "svelte3/svelte3"
		}
	],
	"rules": {
		"prettier/prettier": "error",
		"indent": ["warn", "tab"],
		"quotes": ["warn", "single", { "avoidEscape": true }],
		"no-unused-vars": "warn",
		"no-console": "off",
		"no-tabs": "off",
		"no-restricted-syntax": "off",
		"func-names": "off",
		"quote-props": ["error", "consistent-as-needed"],
		"guard-for-in": "off"
	}
}

```
</details>

```js
export const getColor1 = (obj) =>
	obj.lightness === '-1'
		? 'none'
		: TailwindColors[
				get(settings)[obj.palette === 'accent' ? 'accentColorPalette' : 'baseColorPalette']
		  ][obj.lightness];
```

### What did you expect to happen?

ESLint to be consistent with EsLint Prettier

### What actually happened?

ESLint was inconsistent; see screenshots below

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

![image](https://user-images.githubusercontent.com/30274440/167319730-8d3898aa-8593-404d-8c6e-8bac105d49a2.png)

![image](https://user-images.githubusercontent.com/30274440/167319732-326d9c1d-4873-4bcf-86a0-7526c4415d77.png)

![image](https://user-images.githubusercontent.com/30274440/167319735-1f64282d-20c4-4d71-a910-47ee862a6f0a.png)

**Comment by justingolden21** [comment]
Since I began working on this project again a few days ago, I've run into many of these types of problems. ESLint now complains about quotes and other things and is inconsistent with ESLint Prettier, I get red lines for non-default exports, for ESLint not being able to find imports (in svelte kit) and ESLint thinking that syntax is incorrect such as optional chaining and nullish coalescing which it had no problems with before. Previously, my project was 100% clean, and now I have hundreds of red lines for dozens of new problems... I did find this specific bug interesting, and happily reported it so we can get a rare update to the indentation logic (which I'm sure is very complex) but also if someone would help me determine why my entire project is suddenly red on the same PC with the same packages and same versions and same project and same branch, it would be greatly appreciated. I've wasted five hours on it today alone...

**Comment by justingolden21** [comment]
Another (unrelated) inconsistency:

![image](https://user-images.githubusercontent.com/30274440/167319934-b358bdef-dc0e-4baf-885e-7b1f6568fd54.png)

![image](https://user-images.githubusercontent.com/30274440/167319941-6bd4d19b-173a-4c2c-ac94-5bd2640ba829.png)

**Comment by aladdin-add** [comment]
If you are using prettier, you'll need to disable all stylistic rules.

**Comment by justingolden21** [comment]
I also get this warning:
![image](https://user-images.githubusercontent.com/30274440/179338264-1d9fa1da-8432-458a-b044-61e1f2642085.png)

Here's my code:
```js
export async function get() {
	return {
		headers: {
			'Cache-Control': 'max-age=0, s-maxage=3600',
			'Content-Type': 'application/xml'
		},
		body: `<?xml version="1.0" encoding="UTF-8" ?>
	<urlset xmlns="https://www.sitemaps.org/schemas/sitemap/0.9">
		${pages
			.map(
				(page) =>
					`<url>
			<loc>${appURL}/${page.url}</loc>
			<changefreq>monthly</changefreq>
			<priority>${page.priority ?? 0.5}</priority>
		  </url>`
			)
			.join('')}
	</urlset>`.trim()
	};
}
```

Seems like Prettier and ESLint disagree on this one as well. This is the only file in my (fairly large) project with this yellow underline after formatting.

**Reply by mdjermanovic** [reply]
> Seems like Prettier and ESLint disagree on this one as well.

If you are using Prettier, then only Prettier should control indentation and you should turn off ESLint's `indent` rule and all other rules that might conflict with Prettier. To disable all conflicting rules, you can use `eslint-config-prettier`:

https://github.com/prettier/eslint-config-prettier#eslint-config-prettier

**Reply by justingolden21** [reply]
I'll try that now, thanks! I already have `eslint-plugin-prettier`, weird that I need `eslint-config-prettier` as well. So many packages just to make these two things work together, given that everyone in the planet seems to use them together anyway lol.

**Reply by justingolden21** [reply]
No dice. Just installed it and already have "prettier" in my eslint config. Running lint and format I still get the same problems with my sitemap.xml.js and app.html files respectively

**Reply by justingolden21** [reply]
I read https://dev.to/s2engineers/how-to-make-eslint-work-with-prettier-avoiding-conflicts-and-problems-57pi which seems to be similar to what you're saying, and it's the same setup I've already got.

**Reply by mdjermanovic** [reply]
I checked again your config from the original post. `eslint-config-prettier` has no effect on the `indent` rule because the rule is re-enabled in `"rules"`, which has precedence. You should remove it from there:

```
"rules": {
		"prettier/prettier": "error",
		"indent": ["warn", "tab"], // <-- remove this line
```

**Comment by justingolden21** [comment]
![image](https://user-images.githubusercontent.com/30274440/179622211-ae7fe656-2468-4cc6-80b4-f46e871d7d79.png)

![image](https://user-images.githubusercontent.com/30274440/179622153-204faeb9-8a55-452f-a8fe-399718bd5340.png)
`sitemap.xml.js` after running `npm run lint`: `"lint": "eslint . --ext .js --ignore-path .gitignore ."`

---

![image](https://user-images.githubusercontent.com/30274440/179622372-f2d3802b-7335-4188-b011-6c02e697a095.png)

`app.html` after running `npm run format`: `"format": "prettier --ignore-path .gitignore --check ."`

---

"Fixed" `sitemap.xml.js`:
![image](https://user-images.githubusercontent.com/30274440/179622534-a0c6d211-c3c0-42f0-9580-ca5184115076.png)

"Fixed" `app.html`:
![image](https://user-images.githubusercontent.com/30274440/179622619-1f1553a4-4d18-4947-a927-b3ac205ea87c.png)
